### PR TITLE
Select sample table update

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Modified the checkbox within the samples table header to select the rows on the current page.
Added new `Select All` and `Deselect All` buttons that selects/deselects all table rows.

## Screenshots or screen recordings
image
...
image

## How to set up and validate locally
1. Go to a group or project samples page.
2. Click the `Select All` button.
3. Verify the all the samples are selected and the table footer has been updated.
4. Click the `Deselect All` button.
5. Verify the all the samples are selected and the table footer has been updated.
6. Check the checkbox within the table header.
7. Verify the all the samples on the page are selected and the table footer has been updated.
8. Uncheck the checkbox within the table header.
9. Verify the all the samples on the page are unselected and the table footer has been updated.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by ksierks** [comment]
> I know this is a draft, but I took a look this morning and say that you had started on "Select Page" functionality. Your current solution will work, but you missed a simpler solution.
> 
> Instead of making a server call, the "Select Page" can instead just call a new method in `selection_controller.js` which sets the currently visible rows in the table to checked. The controller already has the `rowSelectionTargets` which contains the currently visible rows so that can be used to update the storage values. This alternate solution reduces a server call, which is always good.

Thanks, I'll make the changes.

File: app/components/samples/table_component.html.erb [hunk 1]
```diff
@@ -20,28 +20,25 @@
...
-                <%= search_form_for(
-                  @q,
-                  url: select_samples_url(**request.query_parameters),
-                  html: { "data-controller": "filters", id: "select-all-form" },
-                  class: "filters align-middle float-left"
-                ) do |f| %>
-                  <input type="hidden" name="format" value="turbo_stream"/>
-
-                  <label for="select-all" class="sr-only"><%= t(:".select_all") %></label>
+                <div
```
Comment (ericenns) [inline app/components/samples/table_component.html.erb:23]: **nitpick (blocking):** Can you simplify this to just a checkbox tag. See below:

```
<%= check_box_tag "select-page",
                title: t(:".select_page"),
                "aria-label": t(:".select_page"),
                data: {
                  action: "input->selection#togglePage",
                  controller: "filters",
                  selection_target: "selectPage",
                },
                class:
                  "w-4 h-4 mr-2.5 text-primary-600 bg-slate-100 border-slate-300 rounded focus:ring-primary-500 dark:focus:ring-primary-600 dark:ring-offset-slate-800 focus:ring-2 dark:bg-slate-700 dark:border-slate-600" %>
```

This change will required the `Ransack::SortComponent` to be updated, specifically the `flex` on line 1 of the html will need to be changed to `inline-flex`.

Additionally `select_all` translation needs removed from `table_component.en.yml` and `table_component.fr.yml` and `select_page` will need to be added.

**Review by ericenns — CHANGES_REQUESTED** [review]
Just a few comments, otherwise this appears to be working as expected

File: app/components/samples/table_component.html.erb (hunk 1 above)
Comment (ericenns) [reply app/components/samples/table_component.html.erb]: The translations for the table component are stored in `app/components/samples/table_component.en.yml` and `app/components/samples/table_component.fr.yml`. Please make your changes there.

Comment (ksierks) [reply app/components/samples/table_component.html.erb]: cca65f5eeb651e901886f699275fed3aa4e8a6f3

File: app/components/samples/table_component.en.yml [hunk 2]
```diff
@@ -9,4 +9,4 @@ en:
   counts:
     samples: Samples
     selected: Selected
-  select_all: Select All
+  select_page: Select / Deselect Samples
```
Comment (ericenns) [inline app/components/samples/table_component.en.yml:12]: **nitpick (blocking):** Can you change this translation to `Select / Deselect visible sample` in both translation files please.

**Review by ericenns — CHANGES_REQUESTED** [review]
One minor translation change, otherwise this is working as expected.

File: app/components/samples/table_component.en.yml (hunk 2 above)
Comment (ksierks) [reply app/components/samples/table_component.en.yml]: 0d1b2486258a47eb15a591b32664b27022d1ca2d

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**